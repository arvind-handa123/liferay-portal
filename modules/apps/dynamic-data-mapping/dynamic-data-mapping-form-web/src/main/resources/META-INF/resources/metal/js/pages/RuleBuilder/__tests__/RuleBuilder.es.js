import RuleBuilder from '../RuleBuilder.es';
import {dom as MetalTestUtil} from 'metal-dom';

let component;

const configView = {
	context: [
		{
			rows: [
				{
					columns: [
						{
							fields: [
								{
									fieldName: 'text1',
									label: 'label text 1'
								}
							]
						}
					]
				}
			]
		}
	],
	rules: [
		{
			actions: [
				{
					action: 'require',
					target: 'text1',
					expression: '[x+2]'
				}
			],
			conditions: [
				{
					operands: [
						{
							type: 'field',
							value: 'value 1'
						},
						{
							type: 'field',
							value: 'value 2'
						}
					],
					operator: 'equals-to'
				}
			],
			['logical-operator']: 'OR',
		}
	],
	mode: 'view'
};

const configEdit = {
	formContext: [
		{
			rows: [
				{
					columns: [
						{
							fields: [
								{
									fieldName: 'text1',
									label: 'label text 1'
								}
							]
						}
					]
				}
			]
		}
	],
	rules: [
		{
			actions: [
				{
					action: 'require',
					target: 'text1'
				}
			],
			conditions: [
				{
					operands: [
						{
							type: 'field',
							value: 'value 1'
						},
						{
							type: 'field',
							value: 'value 2'
						}
					],
					operator: 'equals-to'
				}
			],
			['logical-operator']: 'OR',
			expression: '[x+2]'
		}
	]
};

describe(
	'RuleBuilder',
	() => {
		beforeEach(
			() => {
				jest.useFakeTimers();
			}
		);
		afterEach(
			() => {
				if (component) {
					component.dispose();
				}
			}
		);
		it('should render the list of rules when mode is set to view',
			() => {
				component = new RuleBuilder(configView);
				component.setState({mode: 'view'});
				expect(component).toMatchSnapshot();
			}
		);

		it('should render rule screen editor when click edit button',
			() => {
				MetalTestUtil.enterDocument('<button id="addFieldButton" class="hide"></button>');

				const ruleCardEdit = document.querySelector('.rule-card-edit');

				component = new RuleBuilder(configView);

				MetalTestUtil.triggerEvent(ruleCardEdit, 'click', {});

				jest.runAllTimers();

				expect(component).toMatchSnapshot();
			}
		);

		it('should render rule screen editor when click add rule button',
			() => {
				MetalTestUtil.enterDocument('<button id="addFieldButton" class="hide"></button>');

				const ruleCardEdit = document.querySelector('#addFieldButton');

				component = new RuleBuilder(configView);

				MetalTestUtil.triggerEvent(ruleCardEdit, 'click', {});

				jest.runAllTimers();

				expect(component).toMatchSnapshot();
			}
		);
	}
);