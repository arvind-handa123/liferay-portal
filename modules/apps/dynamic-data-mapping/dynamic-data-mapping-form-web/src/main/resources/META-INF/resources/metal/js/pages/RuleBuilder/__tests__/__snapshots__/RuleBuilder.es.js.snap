// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleBuilder should render rule screen editor when click add rule button 1`] = `
<div class="container">
  <div class="form-builder-rule-builder">
    <h2 class="form-builder-section-title text-default">rule</h2>
    <h4 class="text-default">define-condition-and-action-to-change-fields-and-elements-on-the-form</h4>
    <ul class="liferay-ddm-form-builder-rule-condition-list liferay-ddm-form-rule-builder-timeline timeline ">
      <li class="timeline-item">
        <div class="panel panel-default">
          <div class="flex-container panel-body">
            <div class="h4 panel-title">condition</div>
            <div class="btn-group dropdown" style="">
              <button class="btn btn-default dropdown-toggle dropdown-toggle-operator text-uppercase" data-toggle="dropdown" type="button" disabled="">
                <span class="dropdown-toggle-selected-value">or</span>
                <span class="caret"></span>
              </button>
              <ul class="dropdown-menu">
                <li class="logic-operator text-uppercase" data-logical-operator-value="or">
                  <a href="#">or</a>
                </li>
                <li class="divider"></li>
                <li class="logic-operator text-uppercase" data-logical-operator-value="and">
                  <a href="#">and</a>
                </li>
              </ul>
            </div>
            <div class="timeline-increment">
              <span class="timeline-icon"></span>
            </div>
          </div>
        </div>
      </li>
      <li class="form-builder-rule-condition-container timeline-item" condition-index="0">
        <div class="panel panel-default">
          <div class="flex-container panel-body">
            <div class="form-group-autofit">
              <div class="form-group-item form-group-item-label form-group-item-shrink">
                <h4>
                  <span class="text-truncate-inline">
                    <span class="text-truncate">if</span>
                  </span>
                </h4>
              </div>
              <div class="condition-if form-group-item" condition-if-index="0">
                <div id="fields-rule">
                  <p>Select Field</p>
                </div>
              </div>
              <div class="condition-operator form-group-item" condition-operator-index="0">
                <div id="fields-rule">
                  <p>Select Field</p>
                </div>
              </div>
              <div class="condition-type form-group-item hide" condition-type-index="0">
                <div id="fields-rule">
                  <p>Select Field</p>
                </div>
              </div>
              <div class="condition-type-value form-group-item hide" condition-type-value-index="0">
                <div id="fields-rule">
                  <p>Text Field</p>
                </div>
              </div>
              <div class="condition-type-value-select form-group-item hide" condition-type-value-select-index="0">
                <div id="fields-rule">
                  <p>Select Field</p>
                </div>
              </div>
            </div>
            <div class="timeline-increment">
              <span class="timeline-icon"></span>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`RuleBuilder should render rule screen editor when click edit button 1`] = `
<div class="container">
  <div class="form-builder-rule-builder">
    <h2 class="form-builder-section-title text-default">rule</h2>
    <h4 class="text-default">define-condition-and-action-to-change-fields-and-elements-on-the-form</h4>
    <ul class="liferay-ddm-form-builder-rule-condition-list liferay-ddm-form-rule-builder-timeline timeline ">
      <li class="timeline-item">
        <div class="panel panel-default">
          <div class="flex-container panel-body">
            <div class="h4 panel-title">condition</div>
            <div class="btn-group dropdown" style="">
              <button class="btn btn-default dropdown-toggle dropdown-toggle-operator text-uppercase" data-toggle="dropdown" type="button" disabled="">
                <span class="dropdown-toggle-selected-value">or</span>
                <span class="caret"></span>
              </button>
              <ul class="dropdown-menu">
                <li class="logic-operator text-uppercase" data-logical-operator-value="or">
                  <a href="#">or</a>
                </li>
                <li class="divider"></li>
                <li class="logic-operator text-uppercase" data-logical-operator-value="and">
                  <a href="#">and</a>
                </li>
              </ul>
            </div>
            <div class="timeline-increment">
              <span class="timeline-icon"></span>
            </div>
          </div>
        </div>
      </li>
      <li class="form-builder-rule-condition-container timeline-item" condition-index="0">
        <div class="panel panel-default">
          <div class="flex-container panel-body">
            <div class="form-group-autofit">
              <div class="form-group-item form-group-item-label form-group-item-shrink">
                <h4>
                  <span class="text-truncate-inline">
                    <span class="text-truncate">if</span>
                  </span>
                </h4>
              </div>
              <div class="condition-if form-group-item" condition-if-index="0">
                <div id="fields-rule">
                  <p>Select Field</p>
                </div>
              </div>
              <div class="condition-operator form-group-item" condition-operator-index="0">
                <div id="fields-rule">
                  <p>Select Field</p>
                </div>
              </div>
              <div class="condition-type form-group-item hide" condition-type-index="0">
                <div id="fields-rule">
                  <p>Select Field</p>
                </div>
              </div>
              <div class="condition-type-value form-group-item hide" condition-type-value-index="0">
                <div id="fields-rule">
                  <p>Text Field</p>
                </div>
              </div>
              <div class="condition-type-value-select form-group-item hide" condition-type-value-select-index="0">
                <div id="fields-rule">
                  <p>Select Field</p>
                </div>
              </div>
            </div>
            <div class="timeline-increment">
              <span class="timeline-icon"></span>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`RuleBuilder should render the list of rules when mode is set to view 1`] = `
<div class="container">
  <div class="form-builder-rule-builder-container">
    <h1 class="form-builder-section-title text-default">rules</h1>
    <div class="liferay-ddm-form-rule-rules-list-container">
      <div class="form-builder-rule-list">
        <ul class="ddm-form-body-content form-builder-rule-builder-rules-list tabular-list-group">
          <li class="list-group-item">
            <div class="clamp-horizontal list-group-item-content">
              <p class="form-builder-rule-builder-rule-description text-default">
                <b>if </b>
                <span>field </span>
                <span class="label label-lg label-secondary" data-original-title="label text 1" title="label text 1">
                  <span class="text-truncate-inline">
                    <span class="text-truncate">label text 1</span>
                  </span>
                </span>
                <b class="text-lowercase">
                  <em> is-equal-to </em>
                </b>
                <span>field </span>
                <span class="label label-lg label-secondary" data-original-title="label text 2" title="label text 2">
                  <span class="text-truncate-inline">
                    <span class="text-truncate">label text 2</span>
                  </span>
                </span>
                <br>
                <span>
                  <b>require </b>
                </span>
                <span>field </span>
                <span class="label label-lg label-secondary" data-original-title="label text 1" title="label text 1">
                  <span class="text-truncate-inline">
                    <span class="text-truncate">label text 1</span>
                  </span>
                </span>
              </p>
            </div>
            <div class="list-group-item-field">
              <div class="card-col-field">
                <div class="dropdown dropdown-action">
                  <ul class="dropdown-menu dropdown-menu-right">
                    <li class="rule-card-edit" data-card-id="0">
                      <a href="javascript:;">edit</a>
                    </li>
                    <li class="rule-card-delete" data-card-id="0">
                      <a href="javascript:;">delete</a>
                    </li>
                  </ul>
                  <a class="component-action dropdown-toggle" data-toggle="dropdown" href="#1"></a>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
